src/Forms/ChooseEncoding.cs

```diff
@@ -31,23 +36,22 @@ public void Initialize(string fileName)
         {
             try
             {
-                var file = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
-
-                int length = (int)file.Length;
-                if (length > 100000)
-                    length = 100000;
-
-                file.Position = 0;
-                _fileBuffer = new byte[length];
-                file.Read(_fileBuffer, 0, length);
-
-                for (int i = 0; i < length; i++)
+                using (var file = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
                 {
-                    if (_fileBuffer[i] == 0)
-                        _fileBuffer[i] = 32;
+                    int length = (int)file.Length;
+                    if (length > textBoxPreview.MaxLength)

```

Comment:
> “In code, you can set the value of the Text property to a value that has
> a length greater than the value specified by the MaxLength property.
> This property only affects text entered into the control at run time.”

yeah I know, even in design mode you can, but I think **32767** is alright since it's used just for preview